core/src/main/resources/hudson/PluginManager/advanced.jelly

```diff
@@ -78,9 +78,15 @@ THE SOFTWARE.
                 <h2 class="jenkins-section__title">${%Update Site}</h2>
                 <f:form method="post" action="siteConfigure" name="siteConfigure">
                     <f:entry title="${%URL}">
-                        <f:textbox name="site"
-                                   value="${app.updateCenter.getSite(app.updateCenter.ID_DEFAULT).url}"/>
+                      <j:if test="${app.updateCenter.sites.size() == 1}">
```

Comment:
please only enable this if **all** of the following are true:

1. there exists an `UpdateSite` with an `id` equal to [`hudson.model.UpdateCenter#PREDEFINED_UPDATE_SITE_ID`](https://github.com/jenkinsci/jenkins/blob/46422c0a54371738e395ab49c11d0501527231e4/core/src/main/java/hudson/model/UpdateCenter.java#L166)
2. [`hudson.model.UpdateCenter#ID_DEFAULT`](https://github.com/jenkinsci/jenkins/blob/46422c0a54371738e395ab49c11d0501527231e4/core/src/main/java/hudson/model/UpdateCenter.java#L172) is equal to  [`hudson.model.UpdateCenter#PREDEFINED_UPDATE_SITE_ID`](https://github.com/jenkinsci/jenkins/blob/46422c0a54371738e395ab49c11d0501527231e4/core/src/main/java/hudson/model/UpdateCenter.java#L166)

if either of the above are not true the update site lists have been customized and then the "default" that is being reset to will in most cases not infact be the correct value.

Reply:
actually it is only number 2.  because the save would create the update site with id [`ID_DEFAULT`](https://github.com/jenkinsci/jenkins/blob/46422c0a54371738e395ab49c11d0501527231e4/core/src/main/java/hudson/model/UpdateCenter.java#L166) if not present.